tests/framework/extensions/machinepools/machinepools.go

```diff
@@ -101,3 +162,74 @@ func CreateAllMachinePools(nodeRoles []NodeRoles, machineConfig *v1.SteveAPIObje
 
 	return machinePools
 }
+
+func CreateNewWorkerMachinePool(client *rancher.Client, cluster *v1.SteveAPIObject, updatedCluster *apisV1.Cluster, machineConfig *v1.SteveAPIObject, quantity int32) (error, *v1.SteveAPIObject, *apisV1.Cluster) {
+	nodePoolConfig := []NodeRoles{
+		{
+			ControlPlane: false,
+			Etcd:         false,
+			Worker:       true,
+			Quantity:     1,
+		},
+	}
```

Comment:
I disagree, and also why is this hardcode to 1, but takes a quantity parameter? I feel like we should aim to call it AddMachinePool. I understand wanting to get it done, but doing it this way we're adding tech debt to whoever works on adding other node roles. This would be an easy update as well, by adding a roles parameter

Reply:
I don't understand how this function works tbh. We create a worker machine pool and update the quantity?

Reply:
Correct. This function defines a new worker machine pool and then proceeds to scale based on the defined quantities.

Reply:
I'm fine with `AddMachinePool`, but what did you disagree with @igomez06 ?

Reply:
We can discuss further in our meeting tomorrow, but I am going to push up a change that addresses this concern. In short, I'm utilizing the `nodeRoles` defined in the `provisioningInput`. This way, it's much more dynamic and is not limited to worker.

However you have the `nodeRoles` parameter defined, the scaling will now acknowledge that, I think this is a much better solution.

Reply:
Sweet. I want to push to not even have it come from the node roles from the config, but rather just the test case. I want to keep the nodes roles to be solely for dynamic provisioning test cases. In our meeting I'll make a case why.

Reply:
We met offline regarding this. The overall consensus was to explore having static defined test cases alongside what we have currently with the dynamic `nodeRoles`.

The static test cases remain to be seen if we will do that in a separate PR.

Reply:
For now, I think we can handle static test cases outside of this PR. That's more so expanding the initial groundwork being laid down.